### ISSUE TITLE
BUG: actions/github-script context should be considered dangerous

### ISSUE THREAD

**Issue by gabibguti** [ossf/scorecard#3681]
**Describe the bug**
In GitHub workflows, we can access pull request data in some ways. We currently identify when it's accessed as `github.event.pull_request`.

https://github.com/ossf/scorecard/blob/92470deac34ef9eaaa2707e13b7d800088e0898c/checks/raw/dangerous_workflow.go#L60-L65

When using [actions/github-script](https://github.com/actions/github-script) we can also access this info with `context` object.

```
uses: actions/github-script@v6
        with:
          script: |
            if (context.payload.pull_request.title !== "yoshi-automation") {
              return;
            }
```

This should be considered a dangerous pattern subjectable to [script injection attacks](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#example-of-a-script-injection-attack).

**Reproduction steps**
None.

**Expected behavior**
I expected the following to be considered a dangerous workflow:
https://github.com/googleapis/google-http-java-client/blob/main/.github/workflows/approve-readme.yaml#L23-L36

**Additional context**
None.

**Comment by spencerschrock** [comment]
I'm not sure that's actually dangerous? I think the danger comes when you mix it with bash?

**Comment by gabibguti** [comment]
Agreed. It looks like all possibly dangerous context values for script injection (body, default_branch, email, head_ref, label, message, name, page_name,ref, and title), are only dangerous when executed and not encapsulated in a string. The context object in actions/github-script seems to encapsulate all of these variables as strings, and they become not dangerous when used in the script execution.

**Comment by gabibguti** [comment]
It is still dangerous to:
- execute the strings, for example using [`eval(context.payload.pull_request.title)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval), or incorporating it into the code `$('body').append(context.payload.pull_request.title)`.
- checkout the pull request, for example using `context.payload.pull_request.head.ref`. Currently we look for `github.event.pull_request.head.ref` and other [untrusted checkout patterns within `github.event`](https://github.com/ossf/scorecard/blob/1a17bb812fb2ac23e9d09e86e122f8b67563aed7/checks/raw/dangerous_workflow.go#L30). We should include patterns within `context` too.

**Comment by gabibguti** [comment]
Here's a new workflow example that I believe should be considered a dangerous pattern:
https://github.com/chengjing30/haos/blob/b883b91a8c857a6b85e7981662059403d306a512/.github/workflows/dev.yml#L74-L88
I just wouldn't consider this a dangerous workflow because there's an approval / labeling strategy to mitigate the issue.

**Comment by pnacht** [comment]
Ah yes, that'd be dangerous. The same thing could be done with a standard `run:` step, and it would also be missed by Scorecard.
